miniapps/shifted_methods/distance.cpp

```diff
@@ -0,0 +1,285 @@
+//            MFEM Distance Function Solver - Parallel Version
+//
+// Compile with: make distance
+//
+// Sample runs:
+//   Problem 0: point source.
+//     mpirun -np 4 distance -m ./corners.mesh -rs 3 -t 200.0
+//
+//   Problem 1: zero level set - circle / sphere at the center of the mesh
+//     mpirun -np 4 distance -m ../../data/inline-quad.mesh -rs 3 -o 2 -t 2.0 -p 1
+//     mpirun -np 4 distance -m ../../data/periodic-cube.mesh -rs 2 -o 2 -t 1.0 -p 1
```

Comment:
This sample run (on the periodic cube) was causing me trouble. The solve would complain that the operator was not positive definite and then PCG would fail to converge.  Finally `GLVis` would crash trying to display the results.

I experimented with the options and found that lowering the resolution and increasing the order succeeded. I'm not sure why but perhaps we should adjust this sample run or track down the SPD issue in the operator.

The command line that worked for me was:
```
mpirun -np 4 distance -m ../../data/periodic-cube.mesh -rs 1 -o 3 -t 1.0 -p 1
```

Reply:
I also see some PCG messages with small negative numbers
`PCG: The operator is not positive definite. (Ad, d) = -8.39992e-20`
but it still converges and I can see the `glvis` result.

I changed the solver for this test, it's good to have an examples for the p-laplacian anyway.